keyboards/ash1800/keymaps/default/keymap.c

```diff
@@ -0,0 +1,55 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT( /* Base */
+        KC_ESC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_INS, KC_HOME, KC_PGUP, KC_PSCR,
+                                                                                                       KC_DEL, KC_END, KC_PGDN, KC_SLCK,
+        KC_GRV, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSPC,  KC_NLCK, KC_PSLS, KC_PAST, MO(1),
+        KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, KC_P7, KC_P8, KC_P9, KC_PMNS,
+        KC_CAPS, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_ENT,       KC_P4, KC_P5, KC_P6, KC_PPLS,
+        KC_LSFT, KC_BSLS, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, KC_UP, KC_P1, KC_P2, KC_P3, KC_PENT,
+        KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPC,                  KC_RALT, KC_RCTL, KC_LEFT, KC_DOWN, KC_RIGHT, KC_P0, KC_PDOT
+    ),
+    [1] = LAYOUT(
+        RESET, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,            _______, _______, _______, _______,
+                                                                                                                                      _______, KC_MUTE, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_CALC, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_VOLU, _______, _______, _______, _______,
+        _______, _______, _______,                   KC_MPLY,                                     _______, _______, KC_MPRV, KC_VOLD, KC_MNXT, _______, _______
+    ),
+};
+
+void led_set_user(uint8_t usb_led) {
+    if (IS_LED_ON(usb_led, USB_LED_NUM_LOCK)) {
+        writePinLow(NUM_LOCK_LED_PIN);
+    } else {
+        writePinHigh(NUM_LOCK_LED_PIN);
+    }
+    if (IS_LED_ON(usb_led, USB_LED_CAPS_LOCK)) {
+        writePinLow(CAPS_LOCK_LED_PIN);
+    } else {
+        writePinHigh(CAPS_LOCK_LED_PIN);
+    }
+    if (IS_LED_ON(usb_led, USB_LED_SCROLL_LOCK)) {
+        writePinLow(SCROLL_LOCK_LED_PIN);
+    } else {
+        writePinHigh(SCROLL_LOCK_LED_PIN);
+    }
+}
```

Comment:
There have been some recent improvements to the way lock LEDs are handled. You can replace this function with the following in `ash1800.c`:
```c
bool led_update_kb(led_t led_state) {
    if(led_update_user(led_state)) {
        writePin(NUM_LOCK_LED_PIN, !led_state.num_lock);
        writePin(CAPS_LOCK_LED_PIN, !led_state.caps_lock);
        writePin(SCROLL_LOCK_LED_PIN, !led_state.scroll_lock);
    }
    return true;
}
```

Keymaps can then `return false;` from `led_update_user()`, in order to repurpose the LEDs for something else like layer indication. See #7215 :)

Reply:
do I need to incorporate any commit to do this?